presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -684,17 +679,41 @@ public static void checkCondition(boolean condition, ErrorCodeSupplier errorCode
         }
     }
 
-    public static String annotateColumnComment(String comment, boolean partitionKey)
+    @Nullable
+    public static String annotateColumnComment(Optional<String> comment, boolean partitionKey)
     {
-        comment = nullToEmpty(comment).trim();
+        String normalizedComment = comment.orElse("").trim();
         if (partitionKey) {
-            if (comment.isEmpty()) {
-                comment = "Partition Key";
+            if (normalizedComment.isEmpty()) {
+                normalizedComment = "Partition Key";
             }
             else {
-                comment = "Partition Key: " + comment;
+                normalizedComment = "Partition Key: " + normalizedComment;
+            }
+        }
+        return normalizedComment.isEmpty() ? null : normalizedComment;
+    }
+
+    public static List<String> toPartitionValues(String partitionName)
+    {
+        // mimics Warheouse.makeValsFromName
```

Comment:
Because the Hive one has a very unfriendly API:
* It requires that the caller know the size of the list when calling the function
* It requires that the caller pass in an `AbstractList` (not `List`) as the answer/result of the method

Given the function is simple enough, I find it easier to just rewrite.